### PR TITLE
DSD-921: ButtonGroup component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-921](https://jira.nypl.org/browse/DSD-921)

## This PR does the following:

- Adds the `ButtonGroup` component.

## How has this been tested?

Storybook.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr985-1m3blckrdcbyggbh8j1cpg7wg54fdcpr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6272adc58a4c5796e76a46e9

File: src/components/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        newChildren.push(React.cloneElement(child, { key, isDisabled }));
+      }
+    );
+
+    return (
+      <Stack
+        id={id}
+        className={className}
+        direction={layout}
```
Comment (bigfishdesign13) [inline src/components/ButtonGroup/ButtonGroup.tsx]: Many DS components use a column layout below the `md` breakpoint.  I feel that this component, as well as `CheckboxGroup` and `RadioGroup`, should do the same regardless of how the `layout`prop is set.  What do you think?

File: src/theme/components/buttonGroup.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const ButtonGroup = {
+  baseStyle: ({ buttonWidth }) => ({
+    width: buttonWidth === "default" ? "fit-content" : "100%",
+    button: {
+      flexGrow: buttonWidth === "default" ? "1" : "2",
```
Comment (bigfishdesign13) [inline src/theme/components/buttonGroup.ts]: Why was this necessary?  This style applies to all buttons in the group, so the percentage by which each button will grow and shrink will use the same weighted value.  I pulled the branch locally to experiment.  When I changed to `flexGrow: “1”`, the buttons still sized properly in all cases.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
I left some questions, but ultimately I think they will lead to changes.  Outside of those comments, I think this is great.

File: src/theme/components/buttonGroup.ts (hunk 2 above)
Comment (EdwinGuzman) [reply src/theme/components/buttonGroup.ts]: Ah, I see. This should be `flexGrow: buttonWidth === "default" ? undefined : "1",`. I wasn't sure which was the right one but the "1" for flexGrow makes it expands to the full width. This is how it should work, right?

File: src/components/ButtonGroup/ButtonGroup.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/ButtonGroup/ButtonGroup.tsx]: Yeah, that makes sense. In the ticket I think it said for layout above the md breakpoint and I forgot to ask about that. I think it was meant to say below that breakpoint.

**Comment by EdwinGuzman** [comment]
Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr985-1m3blckrdcbyggbh8j1cpg7wg54fdcpr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6272adc58a4c5796e76a46e9

File: src/components/ButtonGroup/ButtonGroup.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      layout: "row",
+    }}
+  >
+    {(args) => (
+      <ButtonGroup {...args}>
+        <Button buttonType="secondary" id="group-1">
+          Button
+        </Button>
+        <Button id="group-2">Submit</Button>
+        <Button id="group-">Submit</Button>
```
Comment (nypl-wluisi) [inline src/components/ButtonGroup/ButtonGroup.stories.mdx]: think this should be `id="group-3"` ?

**Review by nypl-wluisi — COMMENTED** [review]
@EdwinGuzman This looks pretty good. Can you add some additional tests? Something to test onClick and refs, i.e.,

- "it calls the onClick when a Button component is inside ButtonGroup"
- "it forwards refs to Button component inside ButtonGroup"

File: src/components/ButtonGroup/ButtonGroup.stories.mdx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/ButtonGroup/ButtonGroup.stories.mdx]: Oh yeah, I meant to remove this.

**Comment by EdwinGuzman** [comment]
> Something to test onClick and refs, i.e.,

The `ButtonGroup` is a stylistic wrapper and it doesn't accept refs or event handlers. Those are set in each individual button and you wouldn't want a single `onClick` to be passed to all the `Buttons` in the group. Is this what you were thinking?

I will add a test for the `isDisabled` prop, however.

File: src/theme/components/buttonGroup.ts (hunk 2 above)
Comment (bigfishdesign13) [reply src/theme/components/buttonGroup.ts]: Yes.

File: src/components/ButtonGroup/ButtonGroup.tsx (hunk 1 above)
Comment (bigfishdesign13) [reply src/components/ButtonGroup/ButtonGroup.tsx]: Ah, sorry.  Looking at the description now, I see I wrote `column` where I should have written `row`.  To fix, I made a chance to the ticket.

```
With `mobile-first` in mind, the styles for the `row` layout will kick in for screen widths greater than 600px.
```

**Review by bigfishdesign13 — APPROVED** [review]
Wonderful.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr985-1m3blckrdcbyggbh8j1cpg7wg54fdcpr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6272adc58a4c5796e76a46e9
